specs/Freeze.md

```diff
@@ -0,0 +1,124 @@
+            // When the app window is restored, show the webview
+            // (unless the user has toggle visibility off).
+            if (m_isVisible)
+            {
+                Unfreeze();
+                m_controller->put_IsVisible(TRUE);
+            }
+        }
+    }
+}
+
+void ViewComponent::Freeze()
+{
+    HRESULT hr = webView->TryFreeze(
+        Callback<ICoreWebView2StagingTryFreezeCompletedHandler>(
+            [](HRESULT errorCode, BOOL isSuccessful) -> HRESULT {
+                std::wstringstream formattedMessage;
+                formattedMessage << "TryFreeze result (0x" << std::hex << errorCode
+                                 << ") " << (isSuccessful ? "succeeded" : "failed");
+                MessageBox(nullptr, formattedMessage.str().c_str(), nullptr, MB_OK);
+                return S_OK;
+            })
+            .Get());
+    if (FAILED(hr))
+        ShowFailure(hr, L"Call to TryFreeze failed");
+}
+
+void ViewComponent::Unfreeze()
+{
+    wil::com_ptr<ICoreWebView2Staging2> webView;
+    webView = m_webView.query<ICoreWebView2Staging2>();
+    webView->Unfreeze();
+}
+```
+
+# Remarks
+WebView2 controller must be invisible when the API is called. Otherwise, the
+API fails with `HRESULT_FROM_WIN32(ERROR_INVALID_STATE)`.   
+Freezing is similar to putting a tab into sleeping in browser. Freezing pauses
+WebView script timers and animations, minimizes CPU usage for the associated
+browser renderer process and allows the operating system to reuse the memory that was
+used by the renderer process for other processes.   
+See [Sleeping Tabs FAQ](https://techcommunity.microsoft.com/t5/articles/sleeping-tabs-faq/m-p/1705434)
+for conditions that might prevent WebView from being frozen. In those situations,
+TryFreeze operation will fail and the completed handler will be invoked with isSuccessful as false.   
+WebView will be automatically unfrozen when it becomes visible. Therefore, the app normally doesn't have to call Unfreeze.
+The app can call `Unfreeze` and then `TryFreeze` periodically for an invisibile WebView so that the WebView can sync up with
+latest data to show fresh content when it becomes visible.
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+## Win32 C++
+```IDL
+interface ICoreWebView2_2 : ICoreWebView2 {
+
+  HRESULT TryFreeze([in] ICoreWebView2StagingTryFreezeCompletedHandler* handler);
+
+  HRESULT Unfreeze();
+}
+
+/// The caller implements this interface to receive the TryFreeze result.
+[uuid(00F206A7-9D17-4605-91F6-4E8E4DE192E3), object, pointer_default(unique)]
+interface ICoreWebView2StagingTryFreezeCompletedHandler : IUnknown {
+
+  HRESULT Invoke([in] HRESULT errorCode, [in] BOOL isSuccessful);
+
+}
+```
+## .Net WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    public partial class CoreWebView2
+    {
+        // There are other API in this interface that we are not showing 
+        public Task<int> TryFreezeAsync();
```

Comment:
Should be Task\<bool\> right?

Reply:
It is currently int, should I change it to bool and create a task to track the wrapper change?